docs/component.md

```diff
@@ -3,353 +3,581 @@ id: component
+for example).
+
+In spite of this, the `html` function allows you to add any HTMLElements you wish, wherever you
+want. Here's a simple example:
+
+```javascript
+// render function inside a "my-button" component
+const myElement = document.createElement('button');
+myElement.innerHTML = 'Click me!';
+myElement.addEventListener('click', console.log);
+
+return this.html`
+  <div>${myElement}</div>
+`;
+```
+
+This will result in:
 
 ```html
 <my-button>
-  <div class="fancy-thing">🎰 bar<div>
+  <div><button>Click me!</button></div>
 </my-button>
 ```
 
-### defaultProps
-To give your props a default value you can set the `defaultProps` property of the component.
-
-If you're using typescript (or @babel/plugin-proposal-class-properties for example), you can also
-declare `defaultProps` as a property of your class - you can see an example of this commented below.
+And clicking the button will result in the click event being printed to the console, as expected.
 
+> __📝 Note:__ We also provide the `this.html` function out of the box, should you need it, like so:
 ```javascript
-import Component from '@biotope/element';
+import Component, { html } from '@biotope/element';
+```
 
-class MyButton extends Component {
-  constructor() {
-    super();
-    this.defaultProps = {
-      foo: 'bar',
-    };
-  }
+### Loops
+It is generally a good idea to split your render function into multiple pieces otherwise it may end
+up too messy. You can do this by using `this.html` as many times as you want. Only when the result
+of that function gets returned in `render`, does the DOM get updated. Here is an example of partials
+being built on a loop.
 
-  render() {
-    return this.html`
-      🎰 ${this.props.foo}
-    `;
-  }
-}
+```javascript
+// render function inside a "my-button" component
+const myArray = [1, 2, 3, 4];
+
+return this.html`
+  <div># Here are ${myArray.length} divs on a loop</div>
+  ${myArray.map(number => this.html`
+    <div>Hey! I'm div number ${number}</div>
+  `)}
+  <div># All done!</div>
+`;
+```
 
-MyButton.componentName = 'my-button';
-MyButton.attributes = ['foo'];
-MyButton.register();
+The result would be:
+
+```html
+<my-button>
+  <div># Here are 4 divs on a loop</div>
+  <div>Hey! I'm div number 1</div>
+  <div>Hey! I'm div number 2</div>
+  <div>Hey! I'm div number 3</div>
+  <div>Hey! I'm div number 4</div>
+  <div># All done!</div>
+</my-button>
 ```
 
+## Attributes
+This is the main way of passing information to any component.
+
+Every component has its own attributes. If you have any experience with adding for example
+`class`es, `id`s, `src`s or `data-*` attributes on regular HTMLElements, then you already know how
+to add/modify/remove attributes on `biotope-element`s.
+
+To add or modify an attribute to a `biotope-element` component, you can simply do:
+
 ```html
 <my-button></my-button>
+<script>
+  const myElement = document.getElementsByTagName('my-button');
```

Comment:
`getElementsByTagName` returns an `HTMLCollection`, so `setAttribute()` can't be used directly.

Consider:
`document.querySelector('my-button')`